src/filecoin/FilecoinEngine.ts

```diff
@@ -382,13 +384,10 @@ export class FilecoinEngine extends CurrencyEngine<
       const messages = messagesResponse.messages
       for (const message of messages) {
         const txid = message.cid
-        const idx = this.findTransaction(this.currencyInfo.currencyCode, txid)
 
-        if (idx >= 0) {
-          // Exit early because we reached transaction history from previous
-          // check
-          return
-        }
+        // Exit early because we reached transaction history from previous
+        // check
+        if (this.otherData.knownTxids[txid]) return
```

Comment:
Use

```
if (this.findTransaction('FIL', normalizeAddress(txid)) !== -1) return
```

Reply:
I cannot use this because it's not available after loading the engine.